plato/servers/base.py

```diff
@@ -559,7 +557,9 @@ async def _select_clients(self, for_next_batch=False):
                     # Skip if this sid is currently `training' with reporting clients
                     # or it has already been selected in this round
                     while sid in self.training_sids or sid in self.selected_sids:
-                        client_process_id = client_process_id % self.clients_per_round + 1
+                        client_process_id = (
```

Comment:
Shouldn't this be the (logical) client_id rather than client_process_id?

Reply:
It was physical `client_id` before the change, and we just renamed it to `client_process_id`. 

This while loop is trying to find an available process for the selected logical client because some of them might still be busy in async mode. Previously the processes were identified by indices so it could be obtained from the modulo operation, but now we are now using PIDs, the code here does need to be changed accordingly.